example/lib/main.dart

```diff
@@ -212,18 +212,24 @@ class _HomeState extends State<Home> with WidgetsBindingObserver {
   }
 
   void _startOrStopRecording() async {
-    if (isRecording) {
-      recorderController.reset();
-      final path = await recorderController.stop(false);
-
-      if (path != null) {
-        isRecordingCompleted = true;
-        debugPrint("Recorded file size: ${File(path).lengthSync()}");
-        debugPrint(path);
+    try {
+      if (isRecording) {
+        recorderController.reset();
+
+        final path = await recorderController.stop(false);
+
+        if (path != null) {
+          isRecordingCompleted = true;
+          debugPrint("Recorded file size: ${File(path).lengthSync()}");
+          debugPrint(path);
+        }
+      } else {
+        await recorderController.record();
       }
-    } else {
-      await recorderController.record(path: path);
+    } catch (e) {
+      debugPrint(e.toString());
     }
+
     setState(() {
```

Comment:
you can add this setState in finally block it seems like it doesn't dependent on try catch.